File: BookStoreApi/BookStoreApi/Dtos/LoginResponseDto.cs
using BookStoreApi.Enums;

namespace BookStoreApi.Dtos;

/// <summary>
/// Dto для ответа на вход в профиль.
/// </summary>
public class LoginResponseDto
{
    /// <summary>
    /// Jwt токен доступа.
    /// </summary>
    public string AccessToken { get; set; }

    /// <summary>
    /// Почта пользователя.
    /// </summary>
    public string Email { get; set; }

    /// <summary>
    /// Роль пользователя.
    /// </summary>
    public RolesEnum Role { get; set; }
}

File: frontend/src/app/dtos/login-response.dto.ts
/** Dto для ответа на вход в профиль. */
export class LoginResponseDto
{
    /** Jwt токен доступа. */
    public accessToken: string;

    /** Почта пользователя. */
    public email: string;

    /** Пароль пользователя. */
    public password: string;

    /** Роль пользователя. */
    public role: string;

    constructor(accessToken: string, email: string, password: string, role: string) {
        this.accessToken = accessToken;
        this.email = email;
        this.password = password;
        this.role = role;
    }
}
